src/test/java/com/example/demo/service/DoctorServiceTest.java

```diff
@@ -0,0 +1,326 @@
+                .password("pass")
+                .specialization("Kardiolog")
+                .build();
+        Doctor doctor = buildDoctor();
+        when(doctorRepository.findByEmail(email)).thenReturn(Optional.of(doctor));
+        //when_then
+        DoctorDataIsNullException resultException = assertThrows(DoctorDataIsNullException.class, () -> doctorService.editDoctorByEmail(email, doctorCommand));
+        //then
+        assertEquals("Doctor fields cannot be null", resultException.getMessage());
+        assertEquals(HttpStatus.BAD_REQUEST, resultException.getHttpStatus());
+    }
+
+    @Test
+    void editDoctorByEmail_newEmailIsReserved_exceptionThrown() {
+        //given
+        String email = "test@gmail.com";
+        DoctorCommand doctorCommand = DoctorCommand.builder()
+                .firstName("krzysztof")
+                .lastName("nowak")
+                .email("new@gmail.com")
+                .password("pass")
+                .specialization("Kardiolog")
+                .build();
+        Doctor doctor = buildDoctor();
+        when(doctorRepository.findByEmail(email)).thenReturn(Optional.of(doctor));
+        when(doctorRepository.existsByEmail(email)).thenReturn(true);
+        //when_then
+        DoctorAlreadyExistException resultException = assertThrows(DoctorAlreadyExistException.class, () -> doctorService.editDoctorByEmail(email, doctorCommand));
+        //them
+        assertEquals("Doctor with email: " + doctorCommand.getEmail() + " already exists", resultException.getMessage());
+        assertEquals(HttpStatus.BAD_REQUEST, resultException.getHttpStatus());
+    }
+
+    @Test
+    void addInstitution_doctorExistsInstitutionExists_addInstitutionToDoctor() {
+        //given
+        String email = "test@gmail.com";
+        String id = "1";
+        Doctor doctor = buildDoctor();
+        Institution institution = Institution.builder()
+                .id(1L)
+                .name("szpital")
+                .city("lodz")
+                .zipCode("12123")
+                .street("narutowicza")
+                .buildingNumber("12")
+                .build();
+        Set<Institution> institutions = new HashSet<>();
+        Doctor updatedDoctor = Doctor.builder()
+                .id(1L)
+                .email("test@gmail.com")
+                .password("pass")
+                .firstName("jan")
+                .lastName("kowalski")
+                .specialization("Kardiolog")
+                .institutions(institutions)
+                .build();
+        when(doctorRepository.findByEmail(email)).thenReturn(Optional.of(doctor));
+        when(institutionRepository.findById(Long.valueOf(id))).thenReturn(Optional.of(institution));
+        when(doctorRepository.save(any())).thenReturn(updatedDoctor);
+        //when
+        DoctorDTO resultDoctorDTO = doctorService.addInstitution(email, id);
+        //then
+        assertEquals(1L, resultDoctorDTO.getInstitutionIds().getFirst());
```

Comment:
mozna sie pokusic o wiecej asercji -> to jest przypisanie to moze sprawdzmy czy przypisana instytucja zawiera dobre dane